x-pack/plugins/security_solution/public/common/components/ml/links/create_explorer_link.test.tsx

```diff
@@ -21,7 +22,14 @@ describe('create_explorer_link', () => {
   });
 
   test('it returns expected link', async () => {
-    const ml = { urlGenerator: new MlUrlGenerator({ appBasePath: '/app/ml', useHash: false }) };
+    const urlService = new UrlService({
+      navigate: async () => {},
+      getUrl: async ({ app, path }, { absolute }) => {
+        return `${absolute ? 'http://localhost:8888' : ''}/app/${app}${path}`;
+      },
+    });
+    const locator = urlService.locators.create(new MlLocatorDefinition());
+    const ml = { locator };
```

Comment:
[Nice-to-have] In `x-pack/plugins/security_solution/public/common/components/ml/links/create_explorer_link.test.tsx` and `x-pack/plugins/security_solution/public/common/lib/kibana/kibana_react.mock.ts` there's some code duplication. Would be nice if we could import a stub ml locator (instance or factory) from `ml` plugin instead of creating it in each test in `security_solution`.